lib/model/query/forms.js

```diff
@@ -134,6 +134,7 @@ const createNew = (partial, project) => async ({ Actees, Datasets, FormAttachmen
 
   // Check for xmlFormId collisions with previously deleted forms
   await Forms.checkDeletedForms(partial.xmlFormId, project.id);
+  await Forms.checkDatasetWarnings(parsedDataset);
```

Comment:
I see that the lines above and below this one use `await`, but if `Forms.checkDatasetWarnings()` doesn't need to be async and return a promise, I think it'd be nice to remove the `await`. I think code is often clearer when things can be done sync.

Reply:
I'm stuck on how to not make this return a promise and not await it. It doesn't strictly *need* to be this way but it fits with `checkMeta`, `checkDeletedForms`, and `checkStructuralChange` in checking something and modifying the `context` on the container.  (`checkDeletedForms` makes a DB call though so it does need to be this way.)

Reply:
> I'm stuck on how to not make this return a promise and not await it.

In `checkDatasetWarnings()`, you can remove `async`, then replace `return resolve();` with `return;`. After that, you can remove `await` here.

> It doesn't strictly *need* to be this way but it fits with `checkMeta`, `checkDeletedForms`, and `checkStructuralChange` in checking something and modifying the `context` on the container.  (`checkDeletedForms` makes a DB call though so it does need to be this way.)

`checkDeletedForms()` definitely needs to be async, but I don't think the other two do. `checkMeta()` calls `reject()`, but it could just `throw`. I think it's a little confusing that these functions are async, and I'd love to refactor them sometime. I don't think that time is now though! But it'd be great not to add another instance of this pattern.

Reply:
> I'm stuck on how to not make this return a promise and not await it.

In `checkDatasetWarnings()`, you can remove `async`, then replace `return resolve();` with `return;`. After that, you can remove `await` here.

> It doesn't strictly *need* to be this way but it fits with `checkMeta`, `checkDeletedForms`, and `checkStructuralChange` in checking something and modifying the `context` on the container.  (`checkDeletedForms` makes a DB call though so it does need to be this way.)

`checkDeletedForms()` definitely needs to be async, but I don't think the other two do. `checkMeta()` calls `reject()`, but it could just `throw`. I think it's a little confusing that these functions are async, and I'd love to refactor them sometime. I don't think that time is now though! But it'd be great not to add another instance of this pattern.

Reply:
Following up something we talked about in review: changing this function to not return a promise leads to this error: 

```
TypeError: Cannot read properties of undefined (reading 'pipe')
    at module.<computed> [as checkDatasetWarnings] (/Users/ktuite/Desktop/code/odk/central-backend/lib/model/container.js:31:31)
    at /Users/ktuite/Desktop/code/odk/central-backend/lib/model/query/forms.js:137:9
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async module.exports (/Users/ktuite/Desktop/code/odk/central-backend/test/integration/fixtures/02-forms.js:17:18)
    at async Object.transaction (/Users/ktuite/Desktop/code/odk/central-backend/node_modules/slonik/dist/src/connectionMethods/transaction.js:22:24)
    at async Object.createConnection (/Users/ktuite/Desktop/code/odk/central-backend/node_modules/slonik/dist/src/factories/createConnection.js:97:18)  1) "before all" hook in "{root}"
```

at [these lines in the container](https://github.com/getodk/central-backend/blob/master/lib/model/container.js#L31-L32). I could change these lines to use `?.` but I'll leave it as is for now.